packages/ui5-middleware-livetranspile/lib/livetranspile.js

```diff
@@ -24,60 +24,71 @@ fileNotFoundError.file = ""
  * @returns {function} Middleware function to use
  */
 module.exports = function ({ resources, options }) {
-    const plugins =
-        options.configuration && options.configuration.transpileAsync
-            ? [
+    const config = options.configuration || {}
+    const plugins = config.transpileAsync
+        ? [
+              [
+                  "babel-plugin-transform-async-to-promises",
+                  {
+                      inlineHelpers: true
+                  }
+              ]
+          ]
+        : []
+    const babelConfig = config.babelConfig
+        ? config.babelConfig
+        : {
+              sourceMaps: "inline",
+              plugins,
+              presets: [
                   [
-                      "babel-plugin-transform-async-to-promises",
+                      "@babel/preset-env",
                       {
-                          inlineHelpers: true
+                          targets: {
+                              browsers: "last 2 versions, ie 10-11"
+                          }
                       }
                   ]
               ]
-            : []
-    const babelConfig =
-        options.configuration && options.configuration.babelConfig
-            ? options.configuration.babelConfig
-            : {
-                  sourceMaps: "inline",
-                  plugins,
-                  presets: [
-                      [
-                          "@babel/preset-env",
-                          {
-                              targets: {
-                                  browsers: "last 2 versions, ie 10-11"
-                              }
-                          }
-                      ]
-                  ]
-              }
-
+          }
+    const filePatternConfig = config.filePattern || ".js"
     return (req, res, next) => {
         if (
             req.path &&
             req.path.endsWith(".js") &&
             !req.path.includes("resources/") &&
-            !((options.configuration && options.configuration.excludePatterns) || []).some((pattern) =>
-                req.path.includes(pattern)
-            )
+            !(config.excludePatterns || []).some((pattern) => req.path.includes(pattern))
         ) {
             const pathname = parseurl(req).pathname
+            const pathWithPattern = pathname.replace(".js", filePatternConfig)
```

Comment:
probably not necessary if above comment holds true

Reply:
Thats the actual place where the pattern is needed: for the file lookup of the requested js source